src/evolution_metrics.jl

```diff
@@ -0,0 +1,151 @@
+struct IndicatorEvolutionResults{T<:AbstractFloat}
+    t_indicator::AbstractVector{T}
+    x_indicator::AbstractVector{T}
+    t_evolution::AbstractVector{T}
+    x_evolution::AbstractVector{T}
+    surr_x_evolution::AbstractMatrix{T}
+end
+
+"""
+    indicator_evolution(t, x, indicator, n_surr, surr_method, evolution_metric,
+                        wv_indicator_width, wv_indicator_stride,
+                        wv_evolution_width, wv_evolution_stride)
+
+Computes the `evolution_metric` of an `indicator` for a timeseries (`t`, `x`) and
+`n_surr` surrogates generated by `surr_method`. The windowing parameters for the
+computation of the `indicator` are `wv_indicator_width` and `wv_indicator_stride`.
+The windowing parameters for the computation of the `evolution_metric`, are
+`wv_evolution_width` and `wv_evolution_stride`.
+
+Future dev: take vector of indicators such that surrogates need to be generated only once.
```

Comment:
This is easy to do via multiple dispatch, where `indicator` is a `Vector{Function}`.

Reply:
done!